### ISSUE TITLE
High grad_norm values during training and performance gap with my implementation

### ISSUE THREAD

**Issue by npurson** [LaVi-Lab/VG-LLM#59]
While reproducing your work, I noticed that the grad_norm values are consistently high (often > 10) throughout the training process. For example:

```
  4%|▍         | 200/4656 [1:09:12<25:58:08, 20.98s/it]                                                    
{'loss': 5.5917, 'grad_norm': 32.22332000732422, 'learning_rate': 9.995645168701038e-06, 'epoch': 0.04}
```

This behavior seems unusual, as the value should usually be less than the gradient clipping value of 1.0. Interestingly, while I can reproduce your reported results using your original environment/codebase, I encounter a 2-3% accuracy drop when refactoring the code with the latest dependency versions. In this refactored version, the grad_norm stays relatively low (~0.1).

I suspect the abnormal grad_norm in the original setup might be a key factor in achieving the reported performance. Could you please provide some insights on:

1. Is the large grad_norm expected in your training setup? If so, what's its cause?
2. Are there specific implementation details to align the performance?

Thank you for your time and for sharing this great work!

**Comment by Dlonly904** [comment]
May I ask which dataset you used for training? I am using llava_hound_255k, and the loss has always been over 20

**Comment by npurson** [comment]
> May I ask which dataset you used for training? I am using llava_hound_255k, and the loss has always been over 20

Aligned with the VG-LLM's datasets. The loss over 20 is normal for the original implementation. If you change the value of grad_accum_steps, you will find the loss value is propotional to grad_accum_steps. which is a bug in old version of transformers. But this seems to improve the training performance : (

**Comment by Dlonly904** [comment]
> > 请问你用的是哪个数据集进行训练？我用的是llava_hound_255k，损失一直超过20
> 
> 与VG-LLM的数据集保持一致。超过20的损失在原始实现中是正常的。如果你改变grad_accum_steps的价值，你会发现损失值与grad_accum_steps是相较的。这是旧版变形金刚的一个bug。但这似乎能提升训练表现 ： （

Thank you for your reply，I‘ll try.

**Comment by npurson** [comment]
I find out that this is due to the verision of DeepSpeed. See https://github.com/deepspeedai/DeepSpeed/issues/7749
